Wrappers/Python/cil/framework/labels.py

```diff
@@ -0,0 +1,301 @@
+        """
+        engine_orders = {
+            Backends.ASTRA.value: [cls.CHANNEL.value, cls.VERTICAL.value, \
+                                   cls.ANGLE.value, cls.HORIZONTAL.value],
+            Backends.TIGRE.value: [cls.CHANNEL.value, cls.ANGLE.value, \
+                                   cls.VERTICAL.value, cls.HORIZONTAL.value],
+            Backends.CIL.value: [cls.CHANNEL.value, cls.ANGLE.value, \
+                                 cls.VERTICAL.value, cls.HORIZONTAL.value]
+        }
+
+        dim_order = engine_orders[Backends(engine).value]
+
+        if geometry is None:
+            return dim_order
+        else:
+            return [label for label in dim_order if label in geometry.dimension_labels ]
+
+
+    @classmethod
+    def check_order_for_engine(cls, engine, geometry):
+        """
+        Checks if the order of dimensions is correct for a specific engine and geometry.
+
+        Parameters:
+        ----------
+        engine : str
+            The engine name.
+        geometry : AcquisitionGeometry
+
+        Returns:
+        --------
+        bool
+            True if the order of dimensions is correct
+
+        Raises:
+        -------
+        ValueError
+            If the order of dimensions is incorrect.
+        """
+
+        order_requested = cls.get_order_for_engine(engine, geometry)
+
+        if order_requested == list(geometry.dimension_labels):
+            return True
+        else:
+            raise ValueError(
+                f"Expected dimension_label order {order_requested}, \
+                got {list(geometry.dimension_labels)}.\n\
+                Try using `data.reorder('{engine}')` to permute for {engine}")
+
+class FillTypes(_LabelsBase):
+    """
+    Available fill types for image data.
+
+    Attributes
+    ----------
+    RANDOM ('random'): Fill with random values.
+    RANDOM_INT ('random_int'): Fill with random integers.
+
+    Examples
+    --------
+    data.fill(FillTypes.random)
+    data.fill("random")
+    """
+
+    RANDOM = "random"
+    RANDOM_INT = "random_int"
+
+class UnitsAngles(_LabelsBase):
+    """
+    Available units for angles.
+
+    Attributes
+    ----------
+    DEGREE ('degree'): Degrees.
+    RADIAN ('radian'): Radians.
+
+    Examples
+    --------
+    data.geometry.set_unitangles(angle_data, angle_units=UnitsAngles.DEGREE)
+    data.geometry.set_unit(angle_data, angle_units="degree")
+    """
+
+    DEGREE = "degree"
+    RADIAN = "radian"
+
+class AcquisitionTypes(_LabelsBase):
+    """
+    Available acquisition types.
+
+    Attributes
+    ----------
+    PARALLEL ('parallel'): Parallel beam.
+    CONE ('cone'): Cone beam.
+    """
+
+    PARALLEL = "parallel"
+    CONE = "cone"
+
+class AcquisitionDimensions(_LabelsBase):
+    """
+    Available acquisition dimensions.
+
+    Attributes
+    ----------
+    DIM2 ('2D'): 2D acquisition.
+    DIM3 ('3D'): 3D acquisition.
+    """
+
+    DIM2 = "2D"
+    DIM3 = "3D"
```

Comment:
after discussion with @gfardell... these two could be combined into a single `enum.Flag` using `enum.auto`:

```py
class AcquisitionType(Flag):
    PARALLEL = auto()
    CONE = auto()
    DIM2 = auto()
    DIM3 = auto()
```